### PR TITLE
1495/admin read policies

### LINKED ISSUE #1495: Add admin policies like "resolverread", "realmread"
We should think about adding policies for *reading* resolvers, realms, policies, events.

Currently an administrator can read these settings.

Instead of having action values:

* policywrite
* policydelete
* policyread

we might have another action, that simply says "policy" and 
[x] read
[x] write
[ ] delete

I think we can address this in #1436 with the redesign of the policies.

### PR DESCRIPTION
**Pull request by cornelinux**
This adds read policies to several different administrator settings.

Thus not automatically allowing an administrator to read everything.

Closes #1495

### CODE REVIEW

File: doc/policies/admin.rst [hunk 1]
```diff
@@ -391,12 +391,98 @@ type: bool
...
+
+Allow the administrator to read the smtpserver configuration.
+
+smsgateway_write
+~~~~~~~~~~~~~~~~
+
+type: bool
+
+To be able to define new :ref:`smsgateway` or delete existing ones, the
+administrator needs this rights ``smsgateway_write``.
```
Comment (fredreichbier) [inline doc/policies/admin.rst]: typo: *the right

File: doc/policies/admin.rst [hunk 2]
```diff
@@ -391,12 +391,98 @@ type: bool
...
+eventhandling_read
+~~~~~~~~~~~~~~~~~~
+
+type: bool
+
+Allow the administrator to read event handlers.
+
+.. note:: Currently the policies do not take into account resolvers,
+   or realms. Having the right to read event handlers, will allow the
+   administrator to see all policies.
```
Comment (fredreichbier) [inline doc/policies/admin.rst]: I think this should be *to see all event handlers?

File: doc/policies/admin.rst [hunk 3]
```diff
@@ -391,12 +391,98 @@ type: bool
...
+
+Allow the administrator to read the periodictask definitions.
+
 eventhandling_write
 ~~~~~~~~~~~~~~~~~~~
 
 type: bool
 
-Allow the admiinstrator to configure :ref:`eventhandler`.
+Allow the adminstrator to configure :ref:`eventhandler`.
```
Comment (fredreichbier) [inline doc/policies/admin.rst]: *administrator

File: privacyidea/api/machineresolver.py [hunk 4]
```diff
@@ -49,6 +49,7 @@
 
 @machineresolver_blueprint.route('/', methods=['GET'])
 @log_with(log)
+@prepolicy(check_base_action, request, ACTION.MACHINERESOLVERREAD)
```
Comment (fredreichbier) [inline privacyidea/api/machineresolver.py]: We also have a function ``get_resolver`` in this file, I think we should also add the decorator there?

File: privacyidea/api/periodictask.py [hunk 5]
```diff
@@ -106,6 +106,7 @@ def list_periodic_tasks():
 
 @periodictask_blueprint.route('/<ptaskid>', methods=['GET'])
 @log_with(log)
+@prepolicy(check_base_action, request, ACTION.PERIODICTASKREAD)
```
Comment (fredreichbier) [inline privacyidea/api/periodictask.py]: We should also add this decorator to ``list_periodic_tasks``.

File: privacyidea/lib/policy.py [hunk 6]
```diff
@@ -1242,6 +1254,12 @@ def get_static_policy_definitions(scope=None):
                                             "configuration."),
                                   "group": GROUP.SYSTEM,
                                   'mainmenu': [MAIN_MENU.CONFIG]},
+            ACTION.SYSTEMREAD: {'type': 'bool',
+                                "desc": _("Admin is allowed to read "
+                                          "basic system configuration "
```
Comment (fredreichbier) [inline privacyidea/lib/policy.py]: basic system configuration ~~configuration~~ :)

File: doc/policies/admin.rst (hunk 1 above)
Comment (cornelinux) [reply doc/policies/admin.rst]: will do

File: doc/policies/admin.rst (hunk 2 above)
Comment (cornelinux) [reply doc/policies/admin.rst]: right. Will do.

File: doc/policies/admin.rst (hunk 3 above)
Comment (cornelinux) [reply doc/policies/admin.rst]: So you do not like the adminstrator? Lately I met someone, who was the admin ``strator``.
But if you care, will do! :-)

File: privacyidea/api/machineresolver.py (hunk 4 above)
Comment (cornelinux) [reply privacyidea/api/machineresolver.py]: oh yes. Will do

File: privacyidea/api/periodictask.py (hunk 5 above)
Comment (cornelinux) [reply privacyidea/api/periodictask.py]: Who named the function ``list_periodic_tasks`` in the first place!?
But you are right. Will do.

File: privacyidea/lib/policy.py (hunk 6 above)
Comment (cornelinux) [reply privacyidea/lib/policy.py]: OK. Will do. Thanks!

File: doc/policies/admin.rst (hunk 1 above)
Comment (cornelinux) [reply doc/policies/admin.rst]: done in 003f1a44
(might be a good idea to resolve conversations)

File: doc/policies/admin.rst (hunk 2 above)
Comment (cornelinux) [reply doc/policies/admin.rst]: done in 003f1a44
(might be a good idea to resolve conversations)

File: doc/policies/admin.rst (hunk 3 above)
Comment (cornelinux) [reply doc/policies/admin.rst]: done in 003f1a44
(might be a good idea to resolve conversations)

File: privacyidea/api/machineresolver.py (hunk 4 above)
Comment (cornelinux) [reply privacyidea/api/machineresolver.py]: done in 003f1a44
(might be a good idea to resolve conversations)

File: privacyidea/api/periodictask.py (hunk 5 above)
Comment (cornelinux) [reply privacyidea/api/periodictask.py]: done in 003f1a44
(might be a good idea to resolve conversations)

File: privacyidea/lib/policy.py (hunk 6 above)
Comment (cornelinux) [reply privacyidea/lib/policy.py]: done in 003f1a44
(might be a good idea to resolve conversations)

File: tests/testdata/tmp_directory [hunk 7]
```diff
@@ -1 +1 @@
...
...accountExpires': 131024988000000000, 'homeDirectory': '/home/alice', 'mobile': ['1234', '45678'], 'userPassword': 'alicepw', 'givenName': 'Alice', 'objectGUID': '039b36ef-e7c0-42f3-9bf9-ca6a6c0d4d31', 'email': 'alice@test.com', 'sn': 'Cooper'}}, {'dn': 'cn=bob,ou=example,o=test', 'attributes': {'oid': '3', 'cn': 'bob', 'accountExpires': 9223372036854775807, 'homeDirectory': '/home/bob', 'mobile': '123456', 'userPassword': 'bobpw\xc3\xa9\xc3\xa9\xc3\xa9', 'givenName': 'Robert', 'objectGUID': '039b36ef-e7c0-42f3-9bf9-ca6a6c0d4d77', 'email': 'bob@example.com', 'sn': 'Marley'}}, {'dn': 'cn=manager,ou=example,o=test', 'attributes': {'oid': '1', 'cn': 'manager', 'accountExpires': 9223372036854775807, 'mobile': '123354', 'userPassword': 'ldaptest', 'givenName': 'Corny', 'objectGUID': '039b36ef-e7c0-42f3-9bf9-ca6a6c0d4d54', 'email': 'ck@o', 'sn': 'keule'}}, {'dn': 'cn=k\xc3\xb6lbel,ou=example,o=test', 'attributes': {'someAttr': ['value1', 'value2'], 'accoutnExpires': 9223372036854775807, 'oid': '4', 'cn': 'k\xc3\xb6lbel', 'mobile': '123456', 'userPassword': 'mySecret', 'givenName': 'Cornelius', 'objectGUID': '7dd0533c-afe3-4c6f-b49e-af82eaed045c', 'email': 'cko@o', 'sn': 'K\xc3\xb6lbel'}}]
```
[the remark this answers is not in the document]
Comment (cornelinux) [reply tests/testdata/tmp_directory]: Thanks, i removed the file in fa78f5a3, correct?

[the remark this answers is not in the document]
Comment (fredreichbier) [reply tests/testdata/tmp_directory]: whoops, I think we do need this file, it's also in master:
https://github.com/privacyidea/privacyidea/blob/master/tests/testdata/tmp_directory
but its contents were changed by mistake.
I would simply do
```
git show master:tests/testdata/tmp_directory > tests/testdata/tmp_directory
```
and commit and later squash the commits :)

[the remark this answers is not in the document]
Comment (cornelinux) [reply tests/testdata/tmp_directory]: Would *you* or should I? ;-)

[the remark this answers is not in the document]
Comment (fredreichbier) [reply tests/testdata/tmp_directory]: done, I'll squash & merge this when Travis is finished

**Pull request merged**